core/unit_test/headers_self_contained/CMakeLists.txt

```diff
@@ -13,6 +13,9 @@ if(NOT Kokkos_ENABLE_DEPRECATED_CODE_4 OR Kokkos_ENABLE_DEPRECATION_WARNINGS)
   list(REMOVE_ITEM KOKKOS_CORE_HEADERS "Kokkos_Future.hpp" "Kokkos_TaskScheduler.hpp")
 endif()
 
+# Remove MemoryPool from the list as it uses execution spaces
+list(REMOVE_ITEM KOKKOS_CORE_HEADERS "Kokkos_MemoryPool.hpp")
```

Comment:
I do not understand.
What in these changes makes that the `<Kokkos_MemoryPool.hpp>` header is not self contained any more?

Reply:
Errors look like
```
In file included from /tmp/kokkos/core/unit_test/headers_self_contained/tstHeader.cpp:26:
In file included from tmp/kokkos/core/unit_test/headers_self_contained/tstHeader.cpp:26:
/tmp/kokkos/core/src/Kokkos_MemoryPool.hpp:169:11: error: invalid use of incomplete type 'Kokkos::DefaultHostExecutionSpace' (aka 'Kokkos::Serial')
  169 |           Kokkos::DefaultHostExecutionSpace{}, sb_state_array, m_sb_state_array,
      |           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/tmp/kokkos/core/src/fwd/Kokkos_Fwd_SERIAL.hpp:22:7: note: forward declaration of 'Kokkos::Serial'
   22 | class Serial;  ///< Execution space main process on CPU.
      |       ^
In file included from /tmp/kokkos/core/unit_test/headers_self_contained/tstHeader.cpp:26:
/tmp/kokkos/core/src/Kokkos_MemoryPool.hpp:222:11: error: invalid use of incomplete type 'Kokkos::DefaultHostExecutionSpace' (aka 'Kokkos::Serial')
  222 |           Kokkos::DefaultHostExecutionSpace{}, sb_state_array, m_sb_state_array,
      |           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/tmp/kokkos/core/src/fwd/Kokkos_Fwd_SERIAL.hpp:22:7: note: forward declaration of 'Kokkos::Serial'
   22 | class Serial;  ///< Execution space main process on CPU.
      |       ^
```

Reply:
Do we agree on this or shall we find a different solution to make this PR pass the tests?

Reply:
I think it's fine.